Web Inspector: REGRESSION STP Sources tabs fails to load when hitting a breakpoint

the-chenergy opened the pull request:
#### a80cf05ba528f8bbb7a4a06d95215cf8618ed30c
<pre>
Web Inspector: REGRESSION STP Sources tabs fails to load when hitting a breakpoint
<a href="https://rdar.apple.com/153666768">rdar://153666768</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=294627">https://bugs.webkit.org/show_bug.cgi?id=294627</a>

Reviewed by Devin Rousso.

A SourceCodeTreeElement is not a ResourceTreeElement so it doesn&apos;t have
a `resource` property. This is probably what caused the second uncaught
exception in the bug report, &quot;undefined is not an object (evaluating &apos;this.resource.sourceMaps&apos;)&quot;.

(The suspected regression point was <a href="https://github.com/WebKit/WebKit/pull/45103.)">https://github.com/WebKit/WebKit/pull/45103.)</a>

* Source/WebInspectorUI/UserInterface/Views/SourceCodeTreeElement.js:
(WI.SourceCodeTreeElement.prototype.onpopulate):

Canonical link: <a href="https://commits.webkit.org/296422@main">https://commits.webkit.org/296422@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a644c7b53d68168868fa746d720e74fda6e2aa78

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108411 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28072 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18496 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113620 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58825 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28761 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36622 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82322 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111359 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22808 "Found 2 new test failures: fast/filter-image/clipped-filter.html http/tests/security/cached-svg-image-with-css-cross-domain.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97646 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62757 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22224 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15783 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58346 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92178 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15838 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116741 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35465 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26126 "Found 7 new test failures: http/tests/media/hls/hls-audio-tracks-language.html http/tests/media/hls/hls-webvtt-default.html http/tests/media/hls/hls-webvtt-style.html http/tests/media/hls/range-request-cross-origin.html http/tests/media/hls/track-in-band-multiple-cues.html http/tests/media/hls/track-webvtt-multitracks.html platform/mac/media/encrypted-media/fps-generateRequest.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91353 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35838 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93921 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91153 "Found 3 new API test failures: /WebKitGTK/TestResources:/webkit/WebKitWebResource/get-data-error, /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebView/web-process-crashed, /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23239 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36039 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13803 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31206 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35368 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40905 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35079 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38429 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36762 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3ea32e4b2ace01e5e50a1f0748727cb8054e6560. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46931

the-chenergy commented:
The third uncaught exception from the bug report, `undefined is not an object (evaluating 'this.contentBrowser.currentContentView')` is a bit odd because there's nowhere in the code that we deliberately set the sidebar panel's `contentBrowser` to null. The only explanation is that object's initialization [(here)](https://github.com/WebKit/WebKit/blob/56a9730f146942065c105acb782c30924059b62e/Source/WebInspectorUI/UserInterface/Views/ContentBrowserTabContentView.js#L99) wasn't run, which I'm guessing is a result of the second exception that this patch aims at.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a644c7b53d68168868fa746d720e74fda6e2aa78. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46931

the-chenergy committed:
Web Inspector: REGRESSION STP Sources tabs fails to load when hitting a breakpoint
rdar://153666768
https://bugs.webkit.org/show_bug.cgi?id=294627

Reviewed by Devin Rousso.

A SourceCodeTreeElement is not a ResourceTreeElement so it doesn't have
a `resource` property. This is probably what caused the second uncaught
exception in the bug report, "undefined is not an object (evaluating 'this.resource.sourceMaps')".

(The suspected regression point was https://github.com/WebKit/WebKit/pull/45103.)

* Source/WebInspectorUI/UserInterface/Views/SourceCodeTreeElement.js:
(WI.SourceCodeTreeElement.prototype.onpopulate):

Canonical link: https://commits.webkit.org/296422@main
